snuba/datasets/storages/factory.py

```diff
@@ -168,10 +151,18 @@ def get(self, storage_key: StorageKey) -> Storage:
         return self._all_storages[storage_key]
 
     def get_writable_storage_keys(self) -> list[StorageKey]:
-        return list(self._writable_storages.keys())
+        return [
+            storage_key
+            for storage_key, storage in self._all_storages.items()
+            if type(storage) is WritableTableStorage
+        ]
 
     def get_cdc_storage_keys(self) -> list[StorageKey]:
-        return list(self._cdc_storages.keys())
+        return [
+            storage_key
+            for storage_key, storage in self._all_storages.items()
+            if type(storage) is CdcStorage
```

Comment:
Since our `WritableTableStorage` extends from `ReadableTableStorage`, if we filtered for readable storages using `isinstance(), it would be incorrect because all writable storages are also instances of readable storages. Doing `type(object) is class` does an exact match. 

`isinstance()` still works for `CdcStorage` and `WritableTableStorage` so I can update that.

Reply:
Ok. For non writeable, i would probably still use `not isinstance(WritableTableStorage)` rather than type. That way in case we end up subclassing these anywhere, it still works.